clang/lib/Parse/ParseDeclCXX.cpp

```diff
@@ -1717,6 +1717,7 @@ void Parser::ParseClassSpecifier(tok::TokenKind TagTokKind,
           tok::kw___is_fundamental,
           tok::kw___is_integral,
           tok::kw___is_interface_class,
+          tok::kw___is_layout_compatible,
```

Comment:
I don't think this is necessary. It exists for compatibility with old GCCs that would not have had the notion of layout compatibility 

(and I wonder if we could get rid of the whole switch ultimately @AaronBallman)